osh/cmd_parse.py

```diff
@@ -85,6 +85,16 @@ def _KeywordToken(UP_w):
   return cast(Token, part)
 
 
+def _DelimToken(word, arena):
+    # type: (word_t, Arena) -> Token
+    """
+    TODO: get token directly
+    For cases where we'd want to use `word_.LeftMostSpanForWord` but also want
```

Comment:
I feel like this pattern is a little confusing

One possibility for `((` and `(` is to call it `_OperatorToken(w)` and then assert that `w.tag() == word_e.Token`

Which should really be called `word_e.Operator`, it could be

```
word =
  Operator %Token
  Compound %CompoundWord
```

This doesn't work  for `[[`, because in shell `[` is not an operator token while `(` is, but we could deal with that separately